### PR TITLE
Team Journey cards: GUI support

### PR DESCRIPTION
**Pull request by cjb**
@keybase/picnicsquad CC @keybase/react-hackers @mlsteele 

Here's the latest PR for team journey cards.  There are two cards that still need GUI integration -- the "scroll back to see what you missed" button (which is tricky because that message might not be loaded?) and the "one of your messages is getting a lot of attention" card.

I think we could merge this in so that we can stop having a long-lived branch.  It doesn't need a feature flag because the generation of team journey cards itself is behind a feature flag.  This code won't do anything without receiving a team journey card from the service.

### CODE REVIEW

**Comment by thebearjew** [comment]
@cjb Is it also worth testing the service RPCs on this branch as well?

**Comment by cjb** [comment]
@thebearjew Good question.. I think at the moment it's not necessary because the code already in master displays the card type, and this code just displays cards based on that card type without having any more complex integration, so it doesn't do much more for you.

But it might be a bit easier to work with changing the service while seeing what the GUI's doing here.

File: shared/chat/conversation/messages/cards/team-journey/container.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      break
+    default:
+      console.warn(`Unexpected journey card type: ${props.message.cardType}`)
+      return <Kb.Box2 direction="horizontal" />
+  }
+
+  return props.teamname ? (
+    <TeamJourney actions={actions} image={image} loadTeam={loadTeam} teamname={props.teamname} text={text} />
+  ) : (
+    <Kb.Box2 direction="horizontal" />
```
[the remark this answers is not in the document]
Comment (cjb) [reply shared/chat/conversation/messages/cards/team-journey/container.tsx]: Thanks, switched to `null`

File: shared/chat/conversation/messages/cards/team-journey/index.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  text: string
+}
+
+const TeamJourney = (props: Props) => {
+  // Load the team once on mount for its channel list if required.
+  const {loadTeam, teamname} = props
+  React.useEffect(() => {
+    loadTeam !== null && loadTeam()
+  }, [loadTeam, teamname])
+  return (
```
[the remark this answers is not in the document]
Comment (cjb) [reply shared/chat/conversation/messages/cards/team-journey/index.tsx]: Yup, that's correct!

It's intentional that `otherChannels` isn't a prop in the index.  The abstraction idea was that the container's gonna set text/image/actions and then the index only needs to know how to render those.  The container uses `otherChannels` to set `actions` to have the buttons for those channels, so I don't think index needs to know anything about `otherChannels`.  Maybe I missed something though?

File: shared/chat/conversation/messages/cards/team-journey/index.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        teamname="foo"
+        text="Welcome to the team! Say hi to everyone and introduce yourself."
+      />
+    ))
+    .add('Popular channels', () => (
+      <TeamJourney
+        actions={[
+          {label: '#one', onClick: Sb.action('onGoToChan')},
+          {label: '#two', onClick: Sb.action('onGoToChan')},
+          {label: '#three', onClick: Sb.action('onGoToChan')},
```
[the remark this answers is not in the document]
Comment (cjb) [reply shared/chat/conversation/messages/cards/team-journey/index.stories.tsx]: Thanks, added.

File: shared/chat/conversation/messages/cards/team-journey/index.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        teamname={props.teamname}
+        skipBackground={true}
+        style={styles.avatar}
+      />
+      <Kb.Box2 direction="horizontal" gap="xtiny" fullWidth={true}>
+        <Kb.Text style={styles.teamnameText} type="BodySmallBold">
+          {props.teamname}
+        </Kb.Text>
+        <Kb.Text type="BodyTiny">• System message</Kb.Text>
+      </Kb.Box2>
```
[the remark this answers is not in the document]
Comment (cjb) [reply shared/chat/conversation/messages/cards/team-journey/index.tsx]: Neat, thanks for the tip!  Added baseline to the parent though it still looks slightly off to me, let me know if you have any other ideas.

[the remark this answers is not in the document]
Comment (thebearjew) [reply shared/chat/conversation/messages/cards/team-journey/index.tsx]: @cjb yeah lemme poke around for a second

[the remark this answers is not in the document]
Comment (thebearjew) [reply shared/chat/conversation/messages/cards/team-journey/index.tsx]: @cjb Looks aligned to me actually. Might be the dot making it appear off.

**Pull request merged**